src/realm/object-store/sync/sync_session.cpp

```diff
@@ -1205,23 +1194,30 @@ void SyncSession::shutdown_and_wait()
     m_client.wait_for_session_terminations();
 }
 
-void SyncSession::update_access_token(const std::string& signed_token)
+void SyncSession::update_access_token(std::string_view signed_token)
 {
     util::CheckedUniqueLock lock(m_state_mutex);
-    // We don't expect there to be a session when waiting for access token, but if there is, refresh its token.
-    // If not, the latest token will be seeded from SyncUser::access_token() on session creation.
-    if (m_session) {
-        m_session->refresh(signed_token);
-    }
-    if (m_state == State::WaitingForAccessToken) {
-        become_active();
+    switch (m_state) {
+        case State::Active:
+            m_session->refresh(signed_token);
+            break;
+        case State::WaitingForAccessToken:
+            become_active();
+            break;
+        case State::Paused:
+            // token will be pulled from user when the session is unpaused
+            return;
+        case State::Dying:
+        case State::Inactive:
+            do_revive(std::move(lock));
```

Comment:
I wonder if this is the right thing to do. Wouldn't be a bit odd to revive the session if the user closed it?

Reply:
Paused is the state for "the user has explicitly asked for this to not connect even if it could". `user->log_out()` sets the SyncSession's state to Inactive, so `user->log_in()` should revive sessions in that state.